source/PyMaterialX/PyMaterialXGenShader/PyGenContext.cpp

```diff
@@ -0,0 +1,42 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <PyMaterialX/PyMaterialX.h>
+
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/HwShaderGenerator.h>
+
+namespace py = pybind11;
+namespace mx = MaterialX;
+
+void bindPyGenContext(py::module& mod)
+{
+    py::class_<mx::GenUserData, mx::GenUserDataPtr>(mod, "GenUserData")
+        .def("getSelf", static_cast<mx::GenUserDataPtr (mx::GenUserData::*)()>(&mx::GenUserData::getSelf))
+        .def("asA", static_cast<mx::HwClosureContextPtr (mx::GenUserData::*)()>(&mx::GenUserData::asA))
```

Comment:
I needed to specify asA for each GenUserData type, in this case I only found two types, hope this is ok.